runtime/sema/check_composite_declaration.go

```diff
@@ -1229,8 +1229,21 @@ func (checker *Checker) checkTypeRequirement(
 	for _, nestedCompositeDeclaration := range containerDeclaration.Members.Composites() {
 		nestedCompositeIdentifier := nestedCompositeDeclaration.Identifier.Identifier
 		if nestedCompositeIdentifier == declaredCompositeType.Identifier {
+			// If we detected a second nested composite declaration with the same identifier,
```

Comment:
I think this is not related to `address` fix. Is it intentionally included?

Reply:
Argh, good flag! I accidentally committed this in the wrong PR, removing